src/libstd/sys_common/mutex.rs

```diff
@@ -24,11 +24,15 @@ impl Mutex {
     ///
     /// Behavior is undefined if the mutex is moved after it is
     /// first used with any of the functions below.
+    /// Also, until `init` is called, behavior is undefined if this
+    /// mutex is ever used reentrantly, i.e., `raw_lock` or `try_lock`
+    /// are called by the thread currently holding the lock.
     pub const fn new() -> Mutex { Mutex(imp::Mutex::new()) }
 
     /// Prepare the mutex for use.
     ///
     /// This should be called once the mutex is at a stable memory address.
+    /// Behavior is undefined unless this is called before any other operation.
```

Comment:
Could this be clarified to say something like "If this function is called, behavior is undefined ..." to be clear that it's not undefined behavior to *not* call this method, only to call the method after you've already done other operations.